watertap/property_models/SOP_prop_pack.py

```diff
@@ -0,0 +1,551 @@
+                raise PropertyPackageError(
+                    "\nWhile initializing {sb_name}, the degrees of freedom "
+                    "are {dof}, when zero is required. \nInitialization assumes "
+                    "that the state variables should be fixed and that no other "
+                    "variables are fixed. \nIf other properties have a "
+                    "predetermined value, use the calculate_state method "
+                    "before using initialize to determine the values for "
+                    "the state variables and avoid fixing the property variables."
+                    "".format(sb_name=self.name, dof=dof)
+                )
+
+        # ---------------------------------------------------------------------
+        skip_solve = True  # skip solve if only state variables are present
+        for k in self.keys():
+            if number_unfixed_variables(self[k]) != 0:
+                skip_solve = False
+
+        if not skip_solve:
+            # Initialize properties
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                results = solve_indexed_blocks(opt, [self], tee=slc.tee)
+            init_log.info_high(
+                "Property initialization: {}.".format(idaeslog.condition(results))
+            )
+
+        # ---------------------------------------------------------------------
+        # If input block, return flags, else release state
+        if state_vars_fixed is False:
+            if hold_state is True:
+                return flags
+            else:
+                self.release_state(flags)
+
+    def release_state(self, flags, outlvl=idaeslog.NOTSET):
+        """
+        Method to release state variables fixed during initialisation.
+
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of of logging
+        """
+        # Unfix state variables
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        revert_state_vars(self, flags)
+        init_log.info_high("{} State Released.".format(self.name))
+
+
+# this third class, provides the instructions to create state blocks
+@declare_process_block_class("SopStateBlock", block_class=_SopStateBlock)
+class SopStateBlockData(StateBlockData):
+    def build(self):
+        """Callable method for Block construction."""
+        super(SopStateBlockData, self).build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        # first we create the state variables
+        self.flow_mass_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize={("Liq", "H2O"): 1, ("Liq", "oil"): 0.01},
+            bounds=(1e-15, 100),
```

Comment:
```suggestion
            bounds=(0, None),
```
I think these are the bounds that are traditionally used for flow_mass_phase_comp - unless there is a reason why these should be different

Reply:
I'll add here that, to my recollection, @bknueven 's tweaking of ipopt settings via ipopt-watertap, the default solver that we all use in WaterTAP now, helped enable us to use 0 instead of very small numbers (with regard to the lower bound). With regard to the upper bound, I agree that None is the better way. A user could always reset the upper bound to some desired value, subject to his/her use-case.

Reply:
We should use 0 if we mean 0 -- that is also consistent with the rest of the codebase.

Reply:
I changed the bounds to `(0, None)` as per Marcus' suggestion.